Plugins/Public/playercntl_plugin/PimpShip.cpp

```diff
@@ -287,69 +345,220 @@ namespace PimpShip
 		if (!mapInfo[iClientID].bInPimpDealer || !set_bEnablePimpShip)
 			return false;
 
-		int iHardPointID = ToInt(GetParam(wscParam, ' ', 0));
-		int iSelectedItemID = ToInt(GetParam(wscParam, ' ', 1));
-
-		if (mapInfo[iClientID].mapCurrEquip.find(iHardPointID)
-			== mapInfo[iClientID].mapCurrEquip.end())
+		if (mapInfo[iClientID].mapCurrEquip.empty())
 		{
-			PrintUserCmdText(iClientID, L"ERR Invalid hard point ID");
+			PrintUserCmdText(iClientID, L"ERR Impossible, no hardpoints for lights available");
 			return true;
 		}
 
+		uint beginFrom = 0;
+		uint endAt = 0;
+		uint everyN = 1;
+
+		wstring firstArg = GetParam(wscParam, ' ', 0);
+		wstring secondArg = GetParam(wscParam, ' ', 1);
+
+		uint iSelectedItemID = ToUInt(secondArg);
+
+		if (!iSelectedItemID)
+		{
+			for (map<uint, ITEM_INFO>::iterator it = mapAvailableItems.begin(); it != mapAvailableItems.end(); it++)
+			{
+				if (it->second.wscNickname == secondArg)
+				{
+					iSelectedItemID = it->first;
+					break;
+				}
+			}
+		}
+
 		if (mapAvailableItems.find(iSelectedItemID) == mapAvailableItems.end())
 		{
 			PrintUserCmdText(iClientID, L"ERR Invalid item ID");
 			return true;
 		}
 
-		mapInfo[iClientID].mapCurrEquip[iHardPointID].iArchID = mapAvailableItems[iSelectedItemID].iArchID;
-		return UserCmd_ShowSetup(iClientID, wscCmd, wscParam, usage);
-	}
+		ushort index = 0;
+		for (wstring::iterator it = firstArg.begin(); it != firstArg.end(); it++)
+		{
+			index++;
 
-	bool PimpShip::UserCmd_BuyNow(uint iClientID, const wstring &wscCmd, const wstring &wscParam, const wchar_t *usage)
-	{
-		HK_ERROR err; 
+			if (*it >= '0' && *it <= '9') {}
+			else if (*it == '-')
+			{
+				if (it == firstArg.begin() && it == firstArg.end() - 1)
+				{
+					beginFrom = 1;
+					endAt = mapInfo[iClientID].mapCurrEquip.size();
+				}
+				else if (it == firstArg.end() - 1)
+				{
+					beginFrom = ToUInt(firstArg.substr(0, index));
+					endAt = mapInfo[iClientID].mapCurrEquip.size();
+				}
+				else if (it == firstArg.begin())
+				{
+					beginFrom = 1;
+					endAt = ToUInt(firstArg.substr(index));
 
-		wstring wscCharName = (const wchar_t*) Players.GetActiveCharacterName(iClientID);
+					if (endAt < beginFrom)
+						swap(endAt, beginFrom);
+				}
+				else
+				{
+					beginFrom = ToUInt(firstArg.substr(0, index));
+					endAt = ToUInt(firstArg.substr(index));
 
-		// Check the that player is in a ship dealer.
-		if (!mapInfo[iClientID].bInPimpDealer)
-			return false;
+					if (endAt < beginFrom)
+						swap(endAt, beginFrom);
+				}
 
-		// Charge for the equipment pimp.
-		if (set_iCost > 0)
-		{
-			int iCash = 0;
-			if ((err = HkGetCash(wscCharName, iCash)) != HKE_OK)
+				break;
+			}
+			else if (*it == '*')
+			{
+				if (beginFrom == 0)
+				{
+					endAt = mapInfo[iClientID].mapCurrEquip.size();
+
+					if (it == firstArg.begin())
+					{
+						beginFrom = 1;
+						everyN = ToUInt(firstArg.substr(index));
+					}
+					else if (it == firstArg.end() - 1)
```

Comment:
in the event that this is the first * but it's also the end, shouldn't we just error out right there?

Reply:
The syntax allows to execute the command without 2 arguments, only everyN is non-optional.
*2* = 2* = *2 = 1*2 = 1*2* = *2*10 = 1*2*10
A lot of ways you see.